pkg/apis/core/validation/validation.go

```diff
@@ -1336,8 +1345,10 @@ func ValidatePersistentVolume(pv *core.PersistentVolume) field.ErrorList {
 		allErrs = append(allErrs, field.Required(specPath.Child("capacity"), ""))
 	}
 
-	if _, ok := pv.Spec.Capacity[core.ResourceStorage]; !ok || len(pv.Spec.Capacity) > 1 {
+	if capacityValue, ok := pv.Spec.Capacity[core.ResourceStorage]; !ok || len(pv.Spec.Capacity) > 1 {
 		allErrs = append(allErrs, field.NotSupported(specPath.Child("capacity"), pv.Spec.Capacity, []string{string(core.ResourceStorage)}))
+	} else {
+		allErrs = append(allErrs, ValidatePositiveQuantityValue(capacityValue, specPath.Child("capacity").Key(string(core.ResourceStorage)))...)
```

Comment:
This feels odd, pv.Spec.Capacity is validated a few lines below. Update that instead.